base/bucket.go

```diff
@@ -699,9 +707,13 @@ func GetBucket(spec BucketSpec, callback sgbucket.BucketNotifyFn) (bucket Bucket
 
 	}
 
-	if LogDebugEnabled(KeyBucket) {
+	// Possibly wrap the bucket in a LeakyBucket or a LoggingBucket
+	if spec.LeakyBucketConfig != nil {
+		bucket = NewLeakyBucket(bucket, *spec.LeakyBucketConfig)
```

Comment:
Are we double wrapping here if we've already wrapped a walrus bucket on line 662?  I think the handling for Walrus buckets will go away when we bake vbucket support into Walrus, so it's probably ok having a somewhat hacky workaround for the time being.

Reply:
Fixed by doing a type assertion and only wrapping if it's not already wrapped.  

Also added a double check inside `NewLeakyBucket()` to detect accidental double-wrappings and panic.